### PR TITLE
Fix macOS install.sh by patching sysconfig

### PR DESCRIPTION
**Pull request by damian0815**
On macOS, patch python sysconfig just before creating the venv so that extensions (greenlet & grpcio) can build. 

Ref https://github.com/indygreg/python-build-standalone/issues/103, in particular the solution from @alecthomas posted here: https://github.com/indygreg/python-build-standalone/issues/103#issuecomment-1234942425

To use: checkout, cd into `installer`, run `create_installers.sh`, copy `InvokeAI-mac.zip` into an empty folder outside of your existing invokeAI install, unzip it and run `install.sh`.

### CODE REVIEW

**Review by tildebyte — APPROVED** [review]
:shipit:

**Comment by tildebyte** [comment]
@lstein; Please hold off on getting this into 'main'. I have some other fixes which, together with this, should probably go into a "Z-stream" release. I'm going to open a PR some time today/this evening

**Comment by damian0815** [comment]
this isn't shippable, see #1491

**Review by lstein — APPROVED** [review]
Tildebyte should review this as well.

**Review by tildebyte — APPROVED** [review]

**Comment by tildebyte** [comment]
@damian0815; This can be merged whenever you're comfortable with it

**Pull request merged**
